packages/azidjsclient/index.ts

```diff
@@ -0,0 +1,68 @@
+import { PnPClientStorage, getHashCode } from "@pnp/core/index.js";
+import { Queryable } from "@pnp/queryable/index.js";
+import {
+    AccessToken,
+    AuthorizationCodeCredential,
+    AzureCliCredential,
+    AzurePowerShellCredential,
+    ChainedTokenCredential,
+    ClientAssertionCredential,
+    ClientCertificateCredential,
+    ClientSecretCredential,
+    DefaultAzureCredential,
+    DeviceCodeCredential,
+    EnvironmentCredential,
+    GetTokenOptions,
+    InteractiveBrowserCredential,
+    ManagedIdentityCredential,
+    OnBehalfOfCredential,
+    UsernamePasswordCredential,
+    VisualStudioCodeCredential,
+} from "@azure/identity";
+
+export interface IAzureIdentityToken {
+    token: string;
+    expires: Date;
+}
+
+export type ValidCredential = DefaultAzureCredential | ChainedTokenCredential | EnvironmentCredential |
```

Comment:
Do we want to limit the initial set of credentials we call valid as part of this type? I don't know it well enough, but feels like a subset would apply to the library?

Reply:
Well... we could, but I'm not sure why we would given that this is just a wrapper... If someone has a use case that would work then removing it just creates a false barrier.